modelling/src/main/java/org/axonframework/modelling/StateManager.java

```diff
@@ -0,0 +1,135 @@
+     * @param stateType The type of the state.
+     * @param loader    The function to load state.
+     * @param persister The function to persist state.
+     * @param <I>       The type of id.
+     * @param <T>       The type of state.
+     * @return This instance for fluent interfacing.
+     */
+    default <I, T> StateManager register(
+            @Nonnull Class<I> idType,
+            @Nonnull Class<T> stateType,
+            @Nonnull SimpleEntityLoader<I, T> loader,
+            @Nonnull SimpleEntityPersister<I, T> persister
+    ) {
+        return register(idType, stateType, new SimpleEntityAsyncRepository<>(idType, stateType, loader, persister));
+    }
+
+    /**
+     * Retrieves an entity of the given {@code type} and {@code id}. The {@link CompletableFuture} will resolve to
+     * state, or complete exceptionally if it could not be resolved.
+     * <p>
+     *
+     * @param <T>     The type of state to retrieve.
+     * @param type    The type of state to retrieve.
+     * @param id      The id of the state to retrieve.
+     * @param context The {@link ProcessingContext context} to load the model in.
+     * @return a {@link CompletableFuture} which resolves to the model instance.
+     */
+    @Nonnull
+    default <I, T> CompletableFuture<T> loadEntity(
+            @Nonnull Class<T> type,
+            @Nonnull I id,
+            @Nonnull ProcessingContext context
+    ) {
+        return loadManagedEntity(type, id, context).thenApply(ManagedEntity::entity);
+    }
+
+    /**
+     * Retrieves a {@link ManagedEntity} of the given {@code type} and {@code id}. The {@link CompletableFuture} will
+     * resolve to a {@link ManagedEntity}, or complete exceptionally if it could not be resolved.
+     *
+     * @param type    The type of state to retrieve.
+     * @param id      The id of the state to retrieve.
+     * @param context The {@link ProcessingContext context} to load the model in.
+     * @param <I>     The type of the identifier of the entity.
+     * @param <E>     The type of the entity.
+     * @return a {@link CompletableFuture} which resolves to the model instance.
+     */
+    <I, E> CompletableFuture<ManagedEntity<I, E>> loadManagedEntity(
+            @Nonnull Class<E> type,
+            @Nonnull I id,
+            @Nonnull ProcessingContext context);
+
+    /**
+     * Returns the types of entities that are registered with this {@code StateManager}.
+     * @return the types of entities that are registered with this {@code StateManager}.
+     */
+    Set<Class<?>> registeredTypes();
+
+    /**
+     * Returns the {@link AsyncRepository} for the given {@code type}. Throws an exception if no repository is
+     * registered for the given type.
+     *
+     * @param type The type of the entity.
+     * @param <T> The type of the entity.
+     * @return The {@link AsyncRepository} for the given {@code type}.
+     */
+    <T> AsyncRepository<?, T> repository(Class<T> type);
```

Comment:
I guess the `type` should be `@Nonnull`, right?

Reply:
Yes, added!